test/integration/Aave.t.sol

```diff
@@ -428,6 +436,41 @@ contract AaveIntegrationTest is BaseTest
         assertTrue(MathUtils.within1(lBalance, uint256(lAmountToManageOther)));
     }
 
+    function testGetBalance_AddingAfterLoss(uint256 aAmountToManage1, uint256 aAmountToManage2) public allNetworks allPairs
+    {
+        // assume
+        ConstantProductPair lOtherPair = _createOtherPair();
+        (address lAaveToken, , ) = _dataProvider.getReserveTokensAddresses(USDC);
+        (uint256 lReserve0, uint256 lReserve1, ) = _pair.getReserves();
+        uint256 lReserveUSDC = _pair.token0() == USDC ? lReserve0 : lReserve1;
+        int256 lAmountToManagePair = int256(bound(aAmountToManage1, 1_000_000, lReserveUSDC));
+        int256 lAmountToManageOther = int256(bound(aAmountToManage2, 1_000_000, lReserveUSDC));
+
+        // arrange
+        _manager.adjustManagement(
+            _pair,
+            _pair.token0() == USDC ? lAmountToManagePair : int256(0),
+            _pair.token1() == USDC ? lAmountToManagePair : int256(0)
+        );
+
+        // act - asset manager loses 10% of the assets
+        // todo: how to make manager have a loss? the following is complaining that storage slot can't be found
+        // even if I provide the impl address instead of the proxy address
+        // rewinding in time doesn't work
```

Comment:
My guess is they used a packed slot, which you cannot currently find using stdStorage.

You could overwrite the Aave bytecode with a version that produces losses? E.g. find the market you are targeting and copy its source code. Then just edit the interest calc line to be negative or something. Use `vm.etch` to overwrite the bytecode, skip 30 days, and then receive the loss.

Not sure if you will encounter side effects but could be worth if you don't see a more correct way to have aave trigger a loss (check if there is anyway for a market to even report a loss?)

Reply:
I looked into the docs and code. It seems that loss is not a case that is built into the model of the system. Seems like it's designed for up-only interest bearing. 

I guess forge can't find the storage slot cuz the `balanceOf` and `totalSupply` are functions with computation involved for `AToken`s instead of just simple mappings.

So I suggest, I'll just write the tests for cases to simulate profit and no cases to simulate loss. Sounds good?

Reply:
So I added a unit test case that tests for `adjustManagement` after loss. In the existing suite of unit test cases there are cases that test for syncing (through `burn`) after loss so we're good on that front